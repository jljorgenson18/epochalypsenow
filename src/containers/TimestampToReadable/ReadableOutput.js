import React from 'react';
import PropTypes from 'prop-types';

const ReadableOutput = props => {
  const { date } = props;
  if (!date) {
    return null;
  }
  return <p>{date.format()}</p>;
};

ReadableOutput.propTypes = {
  date: PropTypes.object
};

export default ReadableOutput;
