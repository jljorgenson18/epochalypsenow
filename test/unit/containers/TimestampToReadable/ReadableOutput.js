import React from 'react';
import { expect } from 'chai';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReadableOutput from '~/src/containers/TimestampToReadable/ReadableOutput';

describe('ReadableOutput', () => {
  let mockProps;
  beforeEach(() => {
    mockProps = {
      date: {
        format: sinon.stub().callsFake(() => 'Some date format')
      }
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  it('should be here', async () => {
    expect(ReadableOutput).to.be.ok;
  });

  it('should match snapshot', async () => {
    // Arrange
    // Act
    const wrapper = shallow(<ReadableOutput {...mockProps} />);

    // Assert
    expect(toJson(wrapper)).to.matchSnapshot();
  });

  it('should render a date', async () => {
    // Arrange
    // Act
    const wrapper = shallow(<ReadableOutput {...mockProps} />);

    // Assert
    expect(wrapper.find('p').length).to.equal(1);
  });
});
