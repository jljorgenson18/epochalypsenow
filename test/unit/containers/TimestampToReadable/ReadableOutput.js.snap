// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadableOutput should match snapshot 1`] = `
<p>
  Some date format
</p>
`;
